File: doc-examples/example-jackson-cbor-kotlin/src/main/kotlin/example/BookClient.kt
package example

import io.micronaut.http.annotation.Body
import io.micronaut.http.annotation.Post
import io.micronaut.http.client.annotation.Client
import io.micronaut.serde.cbor.CborMediaTypes

@Client("/books")
interface BookClient {

    @Post(processes = [CborMediaTypes.APPLICATION_CBOR])
    fun save(@Body book: Book): Book
}

File: doc-examples/example-jackson-cbor-python/src/main/python/example/BookClient.py
from abc import ABC, abstractmethod
from typing import Annotated

from micronaut.http.annotation import Body, Post
from micronaut.http.client.annotation import Client
from micronaut.serde.cbor import CborMediaTypes

from example.Book import Book


@Client("/books")
class BookClient(ABC):

    @Post(processes=CborMediaTypes.APPLICATION_CBOR)
    @abstractmethod
    def save(self, book: Annotated[Book, Body]) -> Book:
        pass
